include/pulsar/ProducerConfiguration.h

```diff
@@ -532,6 +533,10 @@ class PULSAR_PUBLIC ProducerConfiguration {
      */
     ProducerAccessMode getAccessMode() const;
 
+    ProducerConfiguration& intercept(const std::initializer_list<ProducerInterceptorPtr> interceptors);
```

Comment:
An `initializer_list` as the parameter is strange. Adding an `addInterceptor` API could be better and easier to use.

```c++
ProducerConfiguration& addInterceptor(const ProducerInterceptorPtr& interceptor);
```

Reply:
This is to make it consistent with [the java API](https://github.com/apache/pulsar/blob/932023da50a124384198778bf0b7daa8bf9a3ba4/pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ProducerBuilder.java#L552).

Using it, the user can easily add the interceptors by this:
```cpp
conf.intercept({interceptor1, interceptor2, interceptor3});
```

Reply:
Yes. I see the advantage.

Reply:
Why not use a `std::vector`?

Reply:
I don't have a strong opinion on it. Just because `initializer_list ` is more often used as the lightweight container type for the input parameter. I could change to using vector anyway.

Reply:
OK. I have decided to use `vector` here.